### PR TITLE
Add ability to create Holiday Schedules in ScheduleRuleset

### LINKED ISSUE #1382: Support holidays in schedule GUI

### LINKED ISSUE #3783: Support Holidays in OpenStudio SDK
- Add method to `setApplyHoliday` for [ScheduleRule](https://openstudio-sdk-documentation.s3.amazonaws.com/cpp/OpenStudio-2.9.0-doc/model/html/classopenstudio_1_1model_1_1_schedule_rule.html) and add its forward translation to IDF.
- Implement method to add start date, duration and `setSpecialDayType` for Holiday in the [RunPeriodControlSpecialDays](https://openstudio-sdk-documentation.s3.amazonaws.com/cpp/OpenStudio-2.9.0-doc/model/html/classopenstudio_1_1model_1_1_run_period_control_special_days.html) and add forward translation to IDF. 

Issue submitted previously for supporting holidays in schedule GUI : https://github.com/NREL/OpenStudio/issues/1382

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

 - Fixes #3783 (and #1382 via companion PR)
 - Companion Pull Request: https://github.com/NREL/OpenStudioApplication/pull/60)
 - Add ability to create Holiday Schedules in ScheduleRuleset

See #3783 for the rationale used, explaining why I chose to add the Holiday schedule to `ScheduleRuleset` instead of `ScheduleRule::applyToHoliday` (TL;DR: if in ScheduleRule, potentially multiple ScheduleRules with applyToHoliday=true could be applied during the same week (this translate to a Schedule:Week:Daily...))

### Pull Request Author

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.

 - [x] Model API Changes / Additions
 - [x] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [x] Model API methods are tested (in `src/model/test`)
 - [x] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [x] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [x] Checked behavior in OpenStudioApplication, adjusted policies as needed (`src/openstudio_lib/library/OpenStudioPolicy.xml`)
 - [x] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [x] All new and existing tests passes
 - [x] If methods have been deprecated, update rest of code to use the new methods: N/A

**Labels:**

 - [x] If change to an IDD file, add the label `IDDChange`
 - [x] If breaking existing API, add the label `APIChange`: N/A
 - [x] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: src/energyplus/ForwardTranslator/ForwardTranslateScheduleRuleset.cpp [hunk 1]
```diff
@@ -148,6 +148,8 @@ boost::optional<IdfObject> ForwardTranslator::translateScheduleRuleset( Schedule
     ScheduleDay defaultDaySchedule = modelObject.defaultDaySchedule();
     ScheduleDay summerDesignDaySchedule = modelObject.summerDesignDaySchedule();
     ScheduleDay winterDesignDaySchedule = modelObject.winterDesignDaySchedule();
+    // ScheduleRuleset is the one carrying the Holiday Schedule
+    ScheduleDay holidayDaySchedule = modelObject.holidaySchedule();
```
Comment (joseph-robertson) [inline src/energyplus/ForwardTranslator/ForwardTranslateScheduleRuleset.cpp]: I say we change "holidayDaySchedule" to "holidaySchedule" for consistency. In the week schedule struct as well.

**Comment by joseph-robertson** [comment]
In general I think this looks good. I need to wait on receiving some example idf file(s) to see if this implementation will accommodate the needs.

File: src/energyplus/ForwardTranslator/ForwardTranslateScheduleRuleset.cpp (hunk 1 above)
Comment (jmarrec) [reply src/energyplus/ForwardTranslator/ForwardTranslateScheduleRuleset.cpp]: No problem

Comment (jmarrec) [reply src/energyplus/ForwardTranslator/ForwardTranslateScheduleRuleset.cpp]: Done

**Comment by jmarrec** [comment]
I need to move the VT into the right block

**Pull request merged**